test/verifiedTests/design/testMultiProductionEnvelope.m

```diff
@@ -0,0 +1,38 @@
+% The COBRAToolbox: testmultiProductionEnvelope.m
+%
+% Purpose:
+%     - test the multiProductionEnvelope function
+%
+% Authors:
+%     - Jacek Wachowiak
+global CBTDIR
+% save the current path
+currentDir = pwd;
+
+% initialize the test
+fileDir = fileparts(which('testMultiProductionEnvelope'));
+cd(fileDir);
+
+% test variables
+model = readCbModel([CBTDIR filesep 'test' filesep 'models' filesep 'ecoli_core_model.mat']);
+biomassRxn = model.rxns(1);
```

Comment:
Why aren't you using the actual biomass reaction for the test?
Also:
The function is supposed to find forced by-products (i.e. excretion fluxes through exchange reactions that are non zero for optimal biomass production). So using anaerobic conditions might be a better option, as under aerobic conditions, there are no required by-products for biomass formation in the ecoli core model.

Overall it might be worth updating the documentation for this function accordingly, and change the minimal amount of byproduct formation that is used as a check for coupling.

Reply:
@jacekwachowiak ?